turbo/jsonrpc/eth_block.go

```diff
@@ -345,6 +342,23 @@ func (api *APIImpl) GetBlockTransactionCountByNumber(ctx context.Context, blockN
 	if err != nil {
 		return nil, err
 	}
+
+	chainConfig, err := api.chainConfig(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	if chainConfig.Bor != nil {
+		borStateSyncTxHash := types.ComputeBorTxHash(blockNum, blockHash)
+		_, ok, err := api._blockReader.EventLookup(ctx, tx, borStateSyncTxHash)
+		if err != nil {
+			return nil, err
```

Comment:
Should we not exit from the function in case of an error when state sync events are missing? We can simply not increment the tx count in that case I believe. Wdyt?

Reply:
`blockReader.EventLookup` returns `ok` to tell us if a state sync transaction exists for a given block or not - if it does we increment, if it doesn't we do not increment and return the number

Reply:
if you are referring to the err handling then I think if there is an err it should be bubbled up all the way to the API